backend/src/AppServer.hs

```diff
@@ -50,10 +52,12 @@ submitReportHandler :: RawGameReport -> AppM SubmitGameReportResponse
 submitReportHandler report = case validateReport report of
   Failure errors -> throwError $ err422 {errBody = show errors}
   Success (RawGameReport {..}) -> runDb $ do
-    -- TODO Logging in this stupid monad
+    logInfoN $ "Processing game between " <> winner <> " and " <> loser <> "."
```

Comment:
The ease/readability of this logging is totally deceptive, would never notice how it removes the lifting-into-monad effort if not for all the background you've given on monad transformer stacks 💡